bl2/src/flash_map.c

```diff
@@ -377,6 +377,26 @@ int flash_area_write(const struct flash_area *area, uint32_t off,
     return 0;
 }
 
+int flash_area_get_sector(const struct flash_area *fa, uint32_t off,
+                          struct flash_sector *sector)
+{
+    if (fa == NULL || sector == NULL) {
+        return -1;
+    }
+
+    if (off >= fa->fa_size) {
+        return -1;
+    }
+
+    sector->fs_off = off & ~(FLASH_AREA_IMAGE_SECTOR_SIZE - 1);
```

Comment:
As I understand it gives an offset of the sector from the beginning of its flash device's storage, however the `struct flash_sector.fs_off` should be an offset value "from the start of its flash area (not device)."

Reply:
I'm wrong, I realised as soon as I wrote down the above because `off` is relative to the flash area..
Please ignore my comment.